packages/core/src/notifications/panelNode.ts

```
@@ -126,15 +130,125 @@ export class NotificationsNode implements TreeNode {
+                }
+                // Show open url option
+                getLogger('notifications').verbose(`opening url for notification: ${notification.id} ...`)
+                await vscode.env.openExternal(vscode.Uri.parse(notification.uiRenderInstructions.onClick.url))
+                break
+            case 'openTextDocument':
+                // Display read-only txt document
+                getLogger('notifications').verbose(`showing txt document for notification: ${notification.id} ...`)
+                await this.showReadonlyTextDocument(notification.uiRenderInstructions.content['en-US'].description)
+                break
+        }
+    }
+
+    /**
+     * Shows a read only txt file for the contect of notification on a side column
+     * It's read-only so that the "save" option doesn't appear when user closes the notification
+     */
+    private async showReadonlyTextDocument(content: string): Promise<void> {
+        try {
+            const tempFilePath = path.join(tempDirPath, 'AWSToolkitNotifications.txt')
+
+            if (await fs.existsFile(tempFilePath)) {
+                // If file exist, make sure it has write permission (0o644)
+                await fs.chmod(tempFilePath, 0o644)
+            }
+
+            await fs.writeFile(tempFilePath, content)
+
+            // Set the file permissions to read-only (0o444)
+            await fs.chmod(tempFilePath, 0o444)
+
+            // Now, open the document
+            const document = await vscode.workspace.openTextDocument(tempFilePath)
+
+            const options: vscode.TextDocumentShowOptions = {
+                viewColumn: vscode.ViewColumn.Beside,
+                preserveFocus: true,
+                preview: true,
+            }
+
+            await vscode.window.showTextDocument(document, options)
+        } catch (error) {
+            throw new ToolkitError(`Error showing text document: ${error}`)
```

Just let the error bubble up. This re-thrown error isn't adding any useful information, and it discards potentially useful info from the original error.